waterloop-site/src/state/teams/reducer.ts

```diff
@@ -0,0 +1,31 @@
+import { Team } from 'api/teams';
+import * as actionTypes from '../action-types';
+
+export interface TeamState {
+  all: Team[];
+}
+
+const initialState: TeamState = {
+  all: [],
+};
+
+interface SetTeams {
```

Comment:
Since you store this as a type and only use that type instead of this interface directly, i think it's better to just define this as `type TeamAction = {...}` and use that, unless it somehow prevents you from doing so.